### PR TITLE
Replace remote cursors

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Frontend

### Description
This PR reworks the realtime communication

### Steps

- [ ] Added implementation
- [ ] Added / updated tests
- [ ] Added / updated documentation
- [ ] Added changelog snippet
- [ ] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: frontend/src/components/editor-page/editor-pane/hooks/yjs/use-sync-realtime-users-to-code-mirror.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { useApplicationState } from '../../../../../hooks/common/use-application-state'
+import { useCodemirrorReferenceContext } from '../../../change-content-context/codemirror-reference-context'
+import type { Cursor } from '../code-mirror-extensions/sync/remote-cursors/remote-cursors-extension'
+import { remoteCursorUpdateEffect } from '../code-mirror-extensions/sync/remote-cursors/remote-cursors-extension'
+import { useEffect } from 'react'
+
+export const useSyncRealtimeUsersToCodeMirror = () => {
```
Comment (ErikMichelson) [inline frontend/src/components/editor-page/editor-pane/hooks/yjs/use-sync-realtime-users-to-code-mirror.ts:12]: ESDoc

File: commons/src/y-doc-sync/y-doc-sync-client.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { ConnectionState } from '../message-transporters/message-transporter.js'
+import { MessageType } from '../message-transporters/message.js'
+import { YDocSync } from './y-doc-sync.js'
+import type { Doc } from 'yjs'
+
+export class YDocSyncClient extends YDocSync {
+  protected bindDocumentSyncMessageEvents(doc: Doc) {
+    super.bindDocumentSyncMessageEvents(doc)
+
+    doc.on('update', (update: Uint8Array, origin: unknown) => {
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply commons/src/y-doc-sync/y-doc-sync-client.ts:15]: Yes. Because anything can be the origin. Also internal objects of yjs

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: backend/src/realtime/realtime-note/realtime-note.spec.ts [hunk 3]
```diff
@@ -51,28 +31,30 @@ describe('realtime note', () => {
 
   it('can connect and disconnect clients', () => {
     const sut = new RealtimeNote(mockedNote, 'nothing');
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply backend/src/realtime/realtime-note/realtime-note.spec.ts:33]: That's a common abbreviation in testing and stands for "system under test". It is used to name the thing that you wanna test.

File: backend/src/realtime/realtime-note/realtime-user-status.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    );
+    this.bindRealtimeUserStateEvents(connection);
+  }
+
+  private createInitialRealtimeUserState(
+    displayName: string,
+    realtimeNote: RealtimeNote,
+  ): RealtimeUser {
+    return {
+      username: displayName,
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply backend/src/realtime/realtime-note/realtime-user-status.ts:31]: Not relevant. Display name is no cryptographical information.

**Pull request closed without merging**